app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -384,11 +385,34 @@ const prepareTxPlanDraft = (txPlanArgs: TxPlanArgs): TxPlanDraft => {
     }
   }
 
+  const prepareDeregisterStakingKeyTx = (
+    txPlanArgs: DeregisterStakingKeyTxPlanArgs
+  ): TxPlanDraft => {
+    const {withdrawals, outputs} = prepareWithdrawalTx({
```

Comment:
Two notes: 
- we probably dont need to withdraw the rewards if they are equal to 0, just a detail tho
- checkout the `validateTxPlan` function in this file ( https://github.com/vacuumlabs/adalite/blob/af5c06eb105725b939e662ec662b0ae0134d0006/app/frontend/wallet/shelley/shelley-transaction-planner.ts#L314) 
It validates whether the total rewards are 0 or higher than the required fee, (this would cause some of the deregistrations to fail if a user had small rewards) 
We could fix this by passing also the `txType` from `txPlanArgs` and don't check for the total rewards amount in case its deregistration.
I cant say I like this very much but its probably a reasonable fix until we refactor the `selectMinimalTxPlan` function to a reasonable state.

Reply:
@PeterBenc related to "We could fix this by passing also the txType from txPlanArgs and don't check for the total rewards amount in case its deregistration.". I discussed this with @xdzurman and he claimed that even though the deposit is greater than the fees, the transaction will not pass as only UTXO counts as your "spending power" not your returned deposit. Btw I also changed this condition first, but then if fails on the other checks (e.g. on mainnet) where you do not have any UTXO & are constructing the transaction. Please let me know if you think it should work otherwise, thanks.

Reply:
Sorry but I am not sure how this relates to deposit or spending power in any way, my only point was that the `validateTxPlan` function throws an error if rewards are lower than the fee. So If a user has rewards `0.01 ADA` and the fee is `0.17 ADA` he basically won't be able to deregister his staking key. 

@rizip1 Probably worth discussing this on a quick call.